compiler/damlc/daml-stdlib-src/DA/Internal/Record.daml

```diff
@@ -12,6 +12,51 @@ import GHC.Types
 import GHC.Tuple() -- reqired for Tuple's themselves
 import GHC.Integer.Type() -- required for Integer primitives
 
+-- | `HasField` gives you getter and setter functions for each record field
+-- automatically.
+--
+-- `HasField x r a` is a typeclass that takes three parameters. The first
+-- parameter `x` is the field name, the second parameter `r` is the record type,
+-- and the last parameter `a` is the type of the field in this record. For
+-- example, if we define a type:
+--
+-- ```
+-- data MyRecord = MyRecord with
+--     foo : Int
+--     bar : Text
+-- ```
+--
+-- Then we get, for free, the following HasField instances:
+--
+-- ```
+-- HasField "foo" MyRecord Int
+-- HasField "bar" MyRecord Text
+-- ```
+--
+-- If we want to get a value using HasField, we can use the `getField` function:
+--
+-- ```
+-- getFoo : MyRecord -> Int
+-- getFoo r = getField @"foo" r
+--
+-- getBar : MyRecord -> Text
+-- getBar = getField @"bar"
+-- ```
+--
+-- Note that this uses the “type application” syntax ( `f @t` ) to specify the
+-- field name, and that the `getBar` definition relies on currying to omit the
```

Comment:
Maybe just eta-expand it? Either a user already knows about currying in which case, they can figure this out themselves. Or they don’t, and this is just confusing.